applications/launchpad/build_images.sh

```diff
@@ -1,28 +1,82 @@
+#   export TBN_FEATURES=avx2
+#   export TL_TAG_URL=local-tarilabs
+#   export TL_TAG_PUSH=0
+#
+
+set -e
+
+# If TL Version is not set, use versions.txt to build with?
+if [ -z "${TL_VERSION}" ]; then
+  source versions.txt
+fi
+
+# Version refers to the base_node, wallet, etc. version - set if they not set
+# applications/tari_app_utilities/Cargo.toml
+TL_VERSION=${TL_VERSION:-0.31.1}
+# https://github.com/xmrig/xmrig/releases
+XMRIG_VERSION=${XMRIG_VERSION:-v6.15.3}
+# https://github.com/monero-project/monero/releases
+MONERO_VERSION=${MONERO_VERSION:-0.17.2.3}
+MONERO_SHA256=${MONERO_SHA256:-8069012ad5e7b35f79e35e6ca71c2424efc54b61f6f93238b182981ba83f2311}
+
+#TBN_ARCH=native
+TBN_ARCH=${TBN_ARCH:-x86-64}
+#TBN_FEATURES=avx2
+TBN_FEATURES=${TBN_FEATURES:-safe}
+
+# Docker tag URL
+TL_TAG_URL=${TL_TAG_URL:-quay.io/tarilabs}
+# Disable docker push if 0
+TL_TAG_PUSH=${TL_TAG_PUSH:-0}
 
 build_image() {
-  echo "Building $1 image v$VERSION.."
-  docker build -f docker_rig/$1 --build-arg ARCH=native --build-arg FEATURES=avx2 --build-arg VERSION=$VERSION $3 $4 -t quay.io/tarilabs/$2:latest ./../..
-  docker tag quay.io/tarilabs/$2:latest quay.io/tarilabs/$2:$VERSION
-  docker push quay.io/tarilabs/$2:latest
-  docker push quay.io/tarilabs/$2:$VERSION
+  echo "Building $1 image v$TL_VERSION ..."
+  docker build -f docker_rig/$1 --build-arg ARCH=${TBN_ARCH} \
+    --build-arg FEATURES=${TBN_FEATURES} --build-arg VERSION=$TL_VERSION $3 $4 \
+    -t ${TL_TAG_URL}/$2:"${TL_VERSION}-${TBN_ARCH}-${TBN_FEATURES}" ./../..
+  if [[ "${TL_TAG_PUSH}" == "1" ]]; then
+    docker push ${TL_TAG_URL}/$2:"${TL_VERSION}-${TBN_ARCH}-${TBN_FEATURES}"
+  fi
 }
 
 build_image base_node.Dockerfile tari_base_node
 build_image console_wallet.Dockerfile tari_console_wallet
 build_image mm_proxy.Dockerfile tari_mm_proxy
 build_image sha3_miner.Dockerfile tari_sha3_miner
 build_image tor.Dockerfile tor
-build_image monerod.Dockerfile monerod
-
-echo "Building XMRig image v$VERSION (XMRig v$XMRIG_VERSION)"
-docker build -f docker_rig/xmrig.Dockerfile --build-arg VERSION=$VERSION --build-arg XMRIG_VERSION=$XMRIG_VERSION -t quay.io/tarilabs/xmrig:latest ./../..
-docker tag quay.io/tarilabs/xmrig:latest quay.io/tarilabs/xmrig:$VERSION
-docker push quay.io/tarilabs/xmrig:latest
-docker push quay.io/tarilabs/xmrig:$VERSION
-
-docker build -f docker_rig/frontail.Dockerfile -t quay.io/tarilabs/frontail ./docker_rig
-docker tag quay.io/tarilabs/frontail:latest quay.io/tarilabs/frontail:$VERSION
-docker push quay.io/tarilabs/frontail:latest
-docker push quay.io/tarilabs/frontail:$VERSION
+
+echo "Building Monerod image v$TL_VERSION (Monerod v$MONERO_VERSION) - $TL_VERSION-v$MONERO_VERSION"
+docker build -f docker_rig/monerod.Dockerfile --build-arg VERSION="$TL_VERSION-v$MONERO_VERSION" \
+  --build-arg MONERO_VERSION=$MONERO_VERSION --build-arg MONERO_SHA256=$MONERO_SHA256  \
+  -t ${TL_TAG_URL}/monerod:"${TL_VERSION}-v$MONERO_VERSION-${TBN_ARCH}-${TBN_FEATURES}" ./../..
```

Comment:
Remove `features` from tag; this is not a Rust build

Reply:
Is this really a cross-platform image? If not, we should not include the `arch` in the tag either.

Reply:
My initial idea was to match builds of third-party docker images with the build of TariLab suite at the time or maybe be able to build set versions, but still match the version to TL suite.  
  
```arch``` is X86 CPU feature set, not even selection between ARM vs X86.  
  
I think getting some of the bits together in a working state is useful and can be moved into more useful parts.